tests/specs/mcd/vat-move-diff-spec.k

```diff
@@ -0,0 +1,212 @@
+requires "verification.k"
+
+module VAT-MOVE-DIFF-SPEC
+  imports VERIFICATION
+
+    // Vat_move-diff
+    rule [Vat.move-diff.pass]:
+      <k> #execute ~> CONTINUATION => #halt ~> CONTINUATION </k>
+      <exit-code> 1 </exit-code>
+      <mode> NORMAL </mode>
+      <schedule> ISTANBUL </schedule>
+      <ethereum>
+        <evm>
+          <output> .ByteArray </output>
+          <statusCode> _ => EVMC_SUCCESS </statusCode>
+          <endPC> _ => ?_ </endPC>
+          <callStack> _VCallStack </callStack>
+          <interimStates> _ </interimStates>
+          <touchedAccounts> _ => ?_ </touchedAccounts>
+          <callState>
+            <program> Vat_bin_runtime </program>
+            <jumpDests> #computeValidJumpDests(Vat_bin_runtime) </jumpDests>
+            <id> ACCT_ID </id>
+            <caller> CALLER_ID </caller>
+            <callData> #abiCallData("move", #address(ABI_src), #address(ABI_dst), #uint256(ABI_rad)) ++ CD => ?_ </callData>
+            <callValue> VCallValue </callValue>
+            <wordStack> .WordStack => ?_ </wordStack>
+            <localMem> .Memory => ?_ </localMem>
+            <pc> 0 => ?_ </pc>
+            <gas> VGas => ?_ </gas>
+            <memoryUsed> 0 => ?_ </memoryUsed>
+            <callGas> _ => ?_ </callGas>
+            <static> false </static>
+            <callDepth> VCallDepth </callDepth>
+          </callState>
+          <substate>
+            <selfDestruct> _VSelfDestruct </selfDestruct>
+            <log> _ => ?_ </log>
+            <refund> _ => ?_ </refund>
+          </substate>
+          <gasPrice> _ </gasPrice>
+          <origin> ORIGIN_ID </origin>
+          <blockhashes> _ </blockhashes>
+          <block>
+            <previousHash> _ </previousHash>
+            <ommersHash> _ </ommersHash>
+            <coinbase> _ </coinbase>
+            <stateRoot> _ </stateRoot>
+            <transactionsRoot> _ </transactionsRoot>
+            <receiptsRoot> _ </receiptsRoot>
+            <logsBloom> _ </logsBloom>
+            <difficulty> _ </difficulty>
+            <number> _BLOCK_NUMBER </number>
+            <gasLimit> _ </gasLimit>
+            <gasUsed> _ </gasUsed>
+            <timestamp> TIME </timestamp>
+            <extraData> _ </extraData>
+            <mixHash> _ </mixHash>
+            <blockNonce> _ </blockNonce>
+            <ommerBlockHeaders> _ </ommerBlockHeaders>
+          </block>
+        </evm>
+        <network>
+          <chainID> VChainId </chainID>
+          <activeAccounts> SetItem(ACCT_ID)
+          SetItem(1)
+          SetItem(2)
+          SetItem(3)
+          SetItem(4)
+          SetItem(5)
+          SetItem(6)
+          SetItem(7)
+          SetItem(8)
+          SetItem(9) _ </activeAccounts>
```

Comment:
Not clear why precompiled accounts have to be filled in like this. They don't seem to be used by the test. Proof side condition correctly have `andBool #notPrecompileAddress(ACCT_ID)`. There are probably other things that could be optimized but this is probably not worth the trouble.

Also, what exactly is this test testing?

Reply:
This spec is largely generated from the ACT specification here (with some tweaks to make it work in our testing harness): https://reports.makerfoundation.com/k-dss/33d35d2934cfc47ccb98/#Vat_move(address,address,uint256)

So I don't want to change this because I want it to be relevant for the MKR folks who want this proof to work. The short answer is that all their proofs are auto-generated using a template which takes into account all the precompiles in this way.

Reply:
Ok then not clear what's the point of review. You can just ask for a formal approval (or I guess you can merge without approval since you are admin)

Reply:
Well I was hoping for you to look over `verification.k` actually, because it add lemmas above and beyond `lemmas.k`, which I would like an audit on. I deleted their `verification.k` file and started from scratch, since theirs had lots of old lemmas which actually weren't used anymore.